### PR TITLE
Implement allow custom animator range

### LINKED ISSUE #242: Allow setting a custom animator range
Required for https://github.com/CartoDB/cartodb.js/issues/786:, _animation for a selected time-series range_

### PR DESCRIPTION
**Pull request by viddo**
Resolves #242 

The steps range have been implicitly `[0, steps]`, to allow setting a custom range I had to make the start+end values explicit. I added a `stepsRange` concept that holds these values, there's a `defaultStepsRange` that works as a fallback in case a `customStepsRange` is not set.

I tried my best to follow existing style and code conventions.

@rochoa can you code review?
cc @javisantana

### CODE REVIEW

File: lib/torque/animator.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 var torque = require('./');
+var StepsRange = require('./animator-steps-range');
```
Comment (javisantana) [inline lib/torque/animator.js]: it's hard to follow when the class is named "AnimatorStepsRange" internally and "StepsRange" here

**Comment by javisantana** [comment]
:es:

File: lib/torque/animator.js (hunk 1 above)
Comment (viddo) [inline lib/torque/animator.js]: Good point, I'll rename it to `AnimatorStepsRange` to match the file/definition then
